partcad/src/partcad/sketch_factory_enrich.py

```
@@ -19,45 +19,62 @@
+
+            if ";" in source_sketch_name:
+                self.source_sketch_name = source_sketch_name.split(";")[0]
+                suffix = source_sketch_name.split(";")[1]
+                self.extra_with = list(map(lambda p: p.split("="), suffix.split(",")))
+            else:
+                self.source_sketch_name = source_sketch_name
+                self.extra_with = []
+
+            self._create(config)
+
+    async def instantiate(self, sketch):
```

Ruff (0.8.2)

`instantiate` is too complex (17 > 10)

(C901)